packages/@aws-cdk/assertions/lib/match.ts

```diff
@@ -1,13 +1,242 @@
+    return new ArrayMatch(pattern, options);
+  }
+
+  /**
+   * Matches the specified pattern to an object found in the same relative path of the target.
+   * The keys and their values (or matchers) must be present in the target but the target can be a superset.
+   * @param pattern the pattern to match
+   * @param options options to configure the matcher
+   */
+  public static objectLike(pattern: {[key: string]: any}, options: ObjectMatchOptions = {}): Match {
+    return new ObjectMatch(pattern, options);
+  }
+
+  /**
+   * Check whether the provided object is a subtype of the `Match` class
+   */
+  public static isMatcher(x: any): x is Match {
+    return x && x instanceof Match;
+  }
+
+  /**
+   * Test whether a target matches the provided pattern.
+   * @param actual the target to match
+   * @return the list of match failures. An empty array denotes a successful match.
+   */
+  public abstract test(actual: any): MatchFailure[];
+}
+
+/**
+ * Options when initializing the `LiteralMatch` class.
+ */
+export interface LiteralMatchOptions {
+  /**
+   * Whether objects nested at any level should be matched partially.
+   * @default false
+   */
+  readonly partialObjects?: boolean;
+}
+
+/**
+ * A Match class that expects the target to match with the pattern exactly.
+ * The pattern may be nested with other matchers that are then deletegated to.
+ */
+export class LiteralMatch extends Match {
+  private readonly partialObjects: boolean;
+
+  constructor(private readonly pattern: any, options: LiteralMatchOptions = {}) {
+    super();
+    this.partialObjects = options.partialObjects ?? false;
+
+    if (Match.isMatcher(this.pattern)) {
+      throw new Error('ExactMatch cannot be nested with another matcher at the top level. Deeper nesting is allowed.');
+    }
+  }
+
+  public test(actual: any): MatchFailure[] {
+    if (Array.isArray(this.pattern)) {
+      return new ArrayMatch(this.pattern, { partial: false }).test(actual);
+    }
+
+    if (typeof this.pattern === 'object') {
+      return new ObjectMatch(this.pattern, { partial: this.partialObjects }).test(actual);
+    }
+
+    if (typeof this.pattern !== typeof actual) {
+      return [{ path: [], message: `Expected type ${typeof this.pattern} but received ${getType(actual)}` }];
+    }
+
+    if (this.pattern === ABSENT) {
+      throw new Error('ABSENT can only be used in an object matcher');
+    }
+
+    if (actual !== this.pattern) {
+      return [{ path: [], message: `Expected ${this.pattern} but received ${actual}` }];
+    }
+
+    return [];
+  }
+}
+
+/**
+ * Options when initializing the `ArrayMatch` class.
+ */
+export interface ArrayMatchOptions {
+  /**
+   * Whether the pattern should partially match with the target.
+   * The target array can contain more elements than expected by the pattern.
+   * Matching elements in the target must be present in the same order as in the pattern.
```

Comment:
Does this describe a subsequence match or a prefix match?

Reply:
(merging comment)

> Call this `subsequence` if so

Reply:
Neither, it means`['a', 'c']` will match `['a', 'b', 'c']`, `['a', 'c']` and `['a', 'x', 'y', 'c']` but not `['c', 'a']`.

I don't think it's either subsequence or prefix here. It's partial or sparse.

Reply:
"greedy ordered subset" might be the term

`['b', 'c']` will not match `['a', 'b', 'b', 'c']` in the current form, correct?

Reply:
If I may quibble, [subsequence](https://en.wikipedia.org/wiki/Subsequence) is definitely the word that describes that property. 

"ordered subset" describes the same thing, but since there's a perfectly cromulent word already, why not use that?

And I think `match([b, c], [a, b, b, c]) == true` ?